src/OpenTelemetry.AutoInstrumentation/Configurations/ConfigurationKeys.cs

```diff
@@ -122,6 +122,18 @@ public static class InstrumentationOptions
             /// Configuration key for GraphQL instrumentation to enable passing query as a document attribute.
             /// </summary>
             public const string GraphQLSetDocument = "OTEL_DOTNET_AUTO_GRAPHQL_SET_DOCUMENT";
+
+            /// <summary>
+            /// Configuration key for SQL Client instrumentation to enable passing text query as a db.statement attribute.
+            /// </summary>
+            public const string SqlClientSetDbStatementForText = "OTEL_DOTNET_AUTO_SQLCLIENT_SET_DBSTATEMENT_FOR_TEXT";
```

Comment:
For the sake of clarity, I'd change `OTEL_DOTNET_AUTO_SQLCLIENT_SET_DBSTATEMENT_FOR_TEXT` to `OTEL_DOTNET_AUTO_SQLCLIENT_SET_DBSTATEMENT_FOR_TEXT_QUERIES` (i.e add `_QUERIES` suffix)

Reply:
I like the change, but it is a bit different than in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/2331#issuecomment-1926638418.

 Fixed in ed0fd0de552bed1026105107116efb0042635e44.

Reply:
I realized instrumentation packages use `FOR_TEXT` suffix in properties names, I'm fine with initial names.

Reply:
I disagree as this is not only for queries.

Reply:
@pellared , see https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Instrumentation.EntityFrameworkCore/Implementation/EntityFrameworkDiagnosticListener.cs#L190-L215

There are 3 types:
 - stored procedures
 - text
 - table direct

We are touching here text.

Same for SQL Client.